core/src/main/java/io/grpc/Context.java

```diff
@@ -520,6 +520,42 @@ public C call() throws Exception {
   }
 
   /**
+   * Wrap an {@link Executor} so that it executes with this context as the {@link #current} context.
+   * It is generally expected that {@link #propagate(Executor)} would be used more commonly than
+   * this method.
+   *
+   * @see #propagate(Executor)
+   */
+  public Executor wrap(final Executor e) {
+    class WrappingExecutor implements Executor {
+      @Override
+      public void execute(Runnable r) {
+        e.execute(wrap(r));
+      }
+    }
+
+    return new WrappingExecutor();
+  }
+
+  /**
+   * Create an executor that propagates the {@link #current} context when {@link Executor#execute}
+   * is called to the {@link #current} context of the {@code Runnable} scheduled. <em>Note that this
+   * is a static method.</em>
+   *
+   * @see #wrap(Executor)
+   */
+  public static Executor propagate(final Executor e) {
+    class PropagatingExecutor implements Executor {
```

Comment:
very very late feedback, but curious. is the method-local type defined here used to name what would otherwise be an anonymous type? ex for debugging reasons, or is there something else I'm missing?